microcosm_pubsub/dispatcher.py

```diff
@@ -62,6 +63,7 @@ def handle_message(self, message, bound_handlers) -> MessageHandlingResult:
                 logger=self.choose_logger(handler),
                 opaque=self.opaque,
             )
+            self.send_metrics(instance)
```

Comment:
I think we probably want this to be invoked one or two levels higher up.

The "clock" starts for an SQS message when we read it (as part of a batch). The clock finishes when we `ack` (delete) or `nack` (extend the timer) after each message. This happens in the logic above, but only for one message.

This means we're delaying the ack/nack call for subsequent messages in a batch on earlier messages writing their metrics.

It'd be better to do so after the entire batch has been processed.